stream.go

```diff
@@ -226,6 +226,12 @@ type StreamReaderOpts struct {
 	// If a value is non-nil, only newer stream entries will be returned.
 	Streams map[string]*StreamEntryID
 
+	// SwitchToNewMessages allows to read unacknowledged messages first and then switch to new ones.
```

Comment:
I'd like to change the naming to `FallbackToUndelivered`, and have the comment be:

```
FallbackToUndelivered will cause any streams in with a non-nil value in Streams to fallback to delivering messages not-yet-delivered to other consumers (as if the value in the Streams map was nil), once the reader has read all its pending messages in the stream. This must be used in conjunction with Group.
```